src/snowflake/snowpark/_internal/code_generation.py

```diff
@@ -0,0 +1,391 @@
+            )  # name could be an alias
+        else:
+            # function/class/variables defined in the same module
+            if inspect.isfunction(obj):
+                func_text += textwrap.dedent(inspect.getsource(obj))
+            elif inspect.isclass(obj):
+                # dynamic class parsing will be handled separately
+                pass
+            else:
+                # c) global variables used by UDF
+                if code_as_comment:
+                    global_vars_text += f"{name}  # variable of type {type(obj)}\n"
+                    continue  # skip the serialization part if we just need code as comment
+                # v does not have source code, then it's a global variable of which the value has been evaluated
+                try:
+                    global_vars_text += f"""
+{name} = pickle.loads(bytes.fromhex('{pickle.dumps(obj).hex()}'))  \
+# {name} is of type {type(obj)} and serialized by snowpark-python
+"""
+                except Exception as exc:
+                    logger.debug(
+                        "Unable to generate source code for object %s of type %s due to exception %r",
+                        type(name),
+                        type(obj),
+                        exc,
+                    )
+                    raise
+
+    # 3. deal with the classes defined in the same module as func's
+    for cls in classes_to_generate:
+        classes_text += textwrap.dedent(inspect.getsource(cls))
+
+    # 4. deal with imports and alias
+    imports = [
+        f"import {name + ' as ' if name != alias else ''}{alias}"
+        for name, alias in to_import
+    ]
+    for module, name_alias_pair in to_import_from_module.items():
+        classes = ", ".join(
+            f"{name + ' as ' if name != alias else ''}{alias}"
+            for name, alias in name_alias_pair
+        )
+        imports.append(f"from {module} import {classes}")
+
+    imports_str = "\n".join(imports) + ("\n" if imports else "")
+    complete_source_code = f"{header_text}{imports_str}{global_vars_text}{classes_text}{func_text}".rstrip()
+
+    # 5. handle func, remove the udf annotation
+    if isinstance(func, FunctionType):
+        func_source_code = remove_function_udf_annotation(
+            textwrap.dedent(inspect.getsource(func))
+        )
+        if not is_lambda(func):
+            complete_source_code = f"{complete_source_code}\n{func_source_code}"
+        func_assignment = (
+            get_lambda_code_text(func_source_code) if is_lambda(func) else func.__name__
+        )
+    else:
+        # BuiltinFunctionType
+        func_assignment = (
+            func.__name__
+            if func_module_name == builtins.__name__
+            else f"{func_module_name}.{func.__name__}"
+        )
+
+    # 6. handle function assignment
+    complete_source_code = f"""\
+{complete_source_code}
+{f"func = {func_assignment}"}\
+""".strip()
+
+    # 7. if code as comment is true, prefix each line with '#'
```

Comment:
I just feel use `"""` is better?

Reply:
I initially want to use `"""`, however, the source code itself code could contain `"""` comments which will mess up the generated code.